File: Systeme-Gestion-Stock/src/app/models/utilisateur.ts


/*
export interface Utilisateur {
  id: number;
  username: string;
  contact: string;
  email: string;
  password: string;
  entrepots_id: number;
  roles_id: number;
  roleName?: string;   // Nom du rôle pour affichage dans la liste
  entrepotName?: string; // Nom de l'entrepôt pour affichage dans la liste
  role?: Role;
  entrepot?: Entrepot;
}

 */

import {Entrepot} from "./entrepot";
import {Role} from "./role";

export interface Utilisateur {
  id?: number;
  username: string;
  contact: string;
  email: string;
  password: string;
  role?: Role;
  entrepot?: Entrepot;

}

File: SpringBoot_Back-End/src/main/java/com/test/entities/Utilisateur.java
package com.test.entities;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

import jakarta.persistence.*;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "utilisateurs")
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties({"authorities"})
public class Utilisateur implements UserDetails {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int Id;

    @Column(name = "username")
    private String username;

    @Column(name = "contact")
    private String contact;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "password", nullable = false)
    private String password;

    @ManyToOne
    @JoinColumn(name = "entrepots_id")
    private Entrepot entrepot;

    /*@ManyToOne
    @JoinColumn(name = "created_by", nullable = true)
    private Utilisateur createdBy;

    @OneToMany(mappedBy = "createdBy", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Utilisateur> createdUsers;*/

    @ManyToOne
    @JoinColumn(name = "roles_id", nullable = false)
    private Role role;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Collections.singletonList(new SimpleGrantedAuthority(this.role.getName()));
    }

    @Override
    public String getUsername() {
        //return email;
        return username;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}
